src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -598,6 +599,18 @@ internal static unsafe void CleanupVariant(IntPtr pDstNativeVariant)
 #endif
         }
 
+        public static unsafe ICustomMarshaler InitializeCustomMarshaller(RuntimeTypeHandle pParameterType, RuntimeTypeHandle pMarshallerType, string cookie, delegate*<string, object> getInstanceMethod)
+        {
+            var marshaller = CustomMarshallerTable.s_customMarshallersTable.GetOrAdd(
+                new CustomMarshallerKey(pParameterType, pMarshallerType, cookie, getInstanceMethod));
+            if (marshaller == null)
+            {
+                throw new ApplicationException();
+            }
```

Comment:
This cannot happen anymore now that this condition is checked in the compiler. Is that right?

Reply:
No, because this is can be only runtime based check. This is checking created marshaller instance. I may express that in IL, but what's the difference?

Reply:
Ah ok